### PR TITLE
Add JKey variant JECDSASecp256k1Key 

### LINKED ISSUE #2428: Add a `JKey` specialization for ECDSA(secp256k1) keys
⚠️ Blocked by https://github.com/hashgraph/hedera-protobufs/issues/110

---

### Problem

To support ECDSA secp256k1 keys, we need a specialization of the abstract `JKey` type that supports secp256k1, say `JECDSAsecp256k1Key`.

### Solution

Update the `JKey` type hierarchy to:

- [x] Add a `JECDSAsecp256k1Key` type whose `isValid()` implementation requires a source `byte[]` of length 33 beginning with exactly `0x02` or `0x03`. (And if necessary for crypto performance, internally maintains both the compressed and uncompressed representations.)
- [x]  Update the gRPC-to-`JKey` ["mapping factory"](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/legacy/core/jproto/JKey.java#L111) to translate a gRPC `Key` object with non-empty `bytes ecdsaSecp256k1` into a `JECDSAsecp256k1Key` instance.
- [x] Update the `JKey`-to-gRPC ["view factory"](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/legacy/core/jproto/JKey.java#L223) to translate a `JECDSAsecp256k1Key` instance to a `Key` whose `oneof` has non-empty `bytes ecdsaSecp256k1`.
- [x] Update the `JKey` serdes (`serialize()` + `deserialize()`) to support this new key type.
- [x] Unit test coverage

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Closes #2428 

1. Add `JECDSASecp256k1Key` that extends `JKey`
2. Updated the `gRPC-to-JKey` and `JKey-to-gRPC` conversions for new key
3. Updated `serialize` and `deserialize` version for new key in `JKey`
4. Added new `JObjectType`,  `JECDSASecp256k1Key`
4. Added unit tests for changes

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/legacy/core/jproto/JObjectType.java [hunk 1]
```diff
@@ -23,39 +23,40 @@
...
+		addLookup(JKey, 15503731);
+		addLookup(JKeyList, 15512048);
+		addLookup(JThresholdKey, 15520365);
+		addLookup(JEd25519Key, 15528682);
+		addLookup(JECDSA_384Key, 15536999);
+		addLookup(JRSA_3072Key, 15620169);
+		addLookup(JContractIDKey, 15545316);
+		addLookup(JFileInfo, 15636803);
+		addLookup(JMemoAdminKey, 15661754);
+        addLookup(JECDSASecp256k1Key, 15661654);
```
Comment (anighanta) [inline hedera-node/src/main/java/com/hedera/services/legacy/core/jproto/JObjectType.java:47]: nit : indendation

**Review by anighanta — APPROVED** [review]
LGTM. indentation changes uncovered some missing test coverage lines.

File: hedera-node/src/main/java/com/hedera/services/legacy/core/jproto/JObjectType.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/src/main/java/com/hedera/services/legacy/core/jproto/JObjectType.java]: Fixed @anighanta . Thanks

File: pom.xml [hunk 2]
```diff
@@ -171,7 +171,7 @@
     <ethereum-core.version>1.12.0-v0.5.0</ethereum-core.version>
     <grpc.version>1.39.0</grpc.version>
     <guava.version>31.0.1-jre</guava.version>
-    <hapi-proto.version>0.20.0-SNAPSHOT</hapi-proto.version>
+    <hapi-proto.version>0.21.0-ecdsa-SNAPSHOT</hapi-proto.version>
```
Comment (tinker-michaelj) [inline pom.xml:174]: We can use `0.21.0-SNAPSHOT` now I believe.

File: hedera-node/src/test/java/com/hedera/services/legacy/unit/serialization/JkeySerializerTest.java [hunk 3]
```diff
@@ -402,4 +414,34 @@ void jKeyProtoSerDes() {
...
+
+	@Test
+	void jKeyECDSASecp256k1KeySerDes() {
+		final Map<String, PrivateKey> pubKey2privKeyMap = new HashMap<>();
+		Key protoKey;
+		JKey jkey = null;
+		try {
+			protoKey = genSingleECDSASecp256k1Key(pubKey2privKeyMap);
+			jkey = JKey.mapKey(protoKey);
+		} catch (DecoderException ignore) {
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hedera-node/src/test/java/com/hedera/services/legacy/unit/serialization/JkeySerializerTest.java]: Agreed . Threw the exceptions in the tests. 😃

File: pom.xml (hunk 2 above)
Comment (Neeharika-Sompalli) [reply pom.xml]: Agreed. Modified it. Thanks Michael !

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Comment by tinker-michaelj** [comment]
Merged into branch https://github.com/hashgraph/hedera-services/tree/secp256k1 for end-to-end testing with SDK support for sig verification.

**Pull request closed without merging**